pkg/domain/infra/tunnel/containers.go

```diff
@@ -509,31 +509,22 @@ func (ic *ContainerEngine) ContainerRun(ctx context.Context, opts entities.Conta
 	}
 	report := entities.ContainerRunReport{Id: con.ID}
 	// Attach
-	if !opts.Detach {
-		err = startAndAttach(ic, con.ID, &opts.DetachKeys, opts.InputStream, opts.OutputStream, opts.ErrorStream)
-		if err == nil {
-			exitCode, err := containers.Wait(ic.ClientCxt, con.ID, nil)
-			if err == nil {
-				report.ExitCode = int(exitCode)
-			}
-		}
-	} else {
+	if opts.Detach {
 		err = containers.Start(ic.ClientCxt, con.ID, nil)
+	} else {
+		err = startAndAttach(ic, con.ID, &opts.DetachKeys, opts.InputStream, opts.OutputStream, opts.ErrorStream)
 	}
 	if err != nil {
 		report.ExitCode = define.ExitCode(err)
-	}
-	if opts.Rm {
-		if err := containers.Remove(ic.ClientCxt, con.ID, bindings.PFalse, bindings.PTrue); err != nil {
-			if errors.Cause(err) == define.ErrNoSuchCtr ||
-				errors.Cause(err) == define.ErrCtrRemoved {
-				logrus.Warnf("Container %s does not exist: %v", con.ID, err)
-			} else {
-				logrus.Errorf("Error removing container %s: %v", con.ID, err)
-			}
+	} else {
+		// Wait
+		var exitCode int32
+		exitCode, err = containers.Wait(ic.ClientCxt, con.ID, nil)
```

Comment:
We should not be doing this if `--detach` was given.

Reply:
Oh man, fair point! I'll polish a bit more